File: 1. Two Sum/.go
func twoSum(nums []int, target int) []int {
    numToIndex := make(map[int]int)

    for i, num := range nums {
        complement := target - num
        if index, found := numToIndex[complement]; found {
            return []int{index, i}
        }
        numToIndex[num] = i
    }

    return []int{}
}

File: 1. Two Sum/.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var numToIndex: [Int: Int] = [:]
        
        for (i, num) in nums.enumerated() {
            let complement = target - num
            if let index = numToIndex[complement] {
                return [index, i]
            }
            numToIndex[num] = i
        }
        
        return []
    }
}
